tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -0,0 +1,374 @@
+                resource_prefix, resource_suffix)
+    security_td.cleanup(force=True)
+    # appnet_td.cleanup(force=True)
+    plain_td.cleanup(force=True)
+
+
+# cleanup_client creates a client runner, and calls its cleanup() method.
+def cleanup_client(project, network, k8s_api_manager, resource_prefix,
+                   resource_suffix, gcp_service_account):
+    runner_kwargs = dict(
+        deployment_name=xds_flags.CLIENT_NAME.value,
+        image_name=xds_k8s_flags.CLIENT_IMAGE.value,
+        td_bootstrap_image=xds_k8s_flags.TD_BOOTSTRAP_IMAGE.value,
+        gcp_project=project,
+        gcp_api_manager=gcp.api.GcpApiManager(),
+        gcp_service_account=gcp_service_account,
+        xds_server_uri=xds_flags.XDS_SERVER_URI.value,
+        network=network,
+        stats_port=xds_flags.CLIENT_PORT.value)
+
+    client_namespace = KubernetesClientRunner.make_namespace_name(
+        resource_prefix, resource_suffix)
+    client_runner = KubernetesClientRunner(
+        k8s.KubernetesNamespace(k8s_api_manager, client_namespace),
+        **runner_kwargs)
+
+    logger.info('Cleanup client')
+    client_runner.cleanup(
+        force=True, force_namespace=True
+    )
+
+
+# cleanup_server creates a server runner, and calls its cleanup() method.
+def cleanup_server(project, network, k8s_api_manager, resource_prefix,
+                   resource_suffix, gcp_service_account):
+    runner_kwargs = dict(
+        deployment_name=xds_flags.SERVER_NAME.value,
+        image_name=xds_k8s_flags.SERVER_IMAGE.value,
+        td_bootstrap_image=xds_k8s_flags.TD_BOOTSTRAP_IMAGE.value,
+        gcp_project=project,
+        gcp_api_manager=gcp.api.GcpApiManager(),
+        gcp_service_account=gcp_service_account,
+        network=network)
+
+    server_namespace = KubernetesServerRunner.make_namespace_name(
+        resource_prefix, resource_suffix)
+    server_runner = KubernetesServerRunner(
+        k8s.KubernetesNamespace(k8s_api_manager, server_namespace),
+        **runner_kwargs)
+
+    logger.info('Cleanup server')
+    server_runner.cleanup(
+        force=True, force_namespace=True
+    )
+
+
+def main(argv):
+    if len(argv) > 1:
+        raise app.UsageError('Too many command-line arguments.')
+    load_keep_config()
+
+    project: str = xds_flags.PROJECT.value
+    network: str = xds_flags.NETWORK.value
+    gcp_service_account: str = xds_k8s_flags.GCP_SERVICE_ACCOUNT.value
+    dry_run: bool = DRY_RUN.value
+
+    # List resources older than KEEP_PERIOD. We only list health-checks and
+    # instance templates because these are leaves in the resource dependency tree.
+    #
+    # E.g. forwarding-rule depends on the target-proxy. So leaked
+    # forwarding-rule indicates there's a leaked target-proxy (because this
+    # target proxy cannot deleted unless the forwarding rule is deleted). The
+    # leaked target-proxy is guaranteed to be a super set of leaked
+    # forwarding-rule.
```

Comment:
I gave this idea another thought. There might be an situation: the target-proxy and forwarding-rule of a url-map is deleted, but not the url-map itself. Then, the standalone url-map will be skipped by this script.

Reply:
forwarding-rule and target-proxy were to explain the relationship between resources.

What we check is health-check, I believe url-map is covered (url-map --> backend-servce --> health-check)